src/pages/progress/index.tsx

```diff
@@ -48,26 +53,101 @@ const WarningSection: FC = () => (
   </section>
 );
 
-const ProgressContent: FC<{ currentPhaseIndex: number }> = ({ currentPhaseIndex }) => (
-  <Box className="flex flex-col items-center justify-center mt-4">
-    <div className="flex flex-col items-center justify-center max-w-[400px]">
-      <span className="w-10 text-blue-700 loading loading-spinner"></span>
-      <h1 className="my-3 text-xl font-bold text-blue-700">Your transaction is in progress.</h1>
-      <WarningSection />
-      <div className="h-0.5 m-auto w-full bg-blue-700 mt-3 mb-1" />
-      <ProgressSteps currentPhaseIndex={currentPhaseIndex} />
-    </div>
-  </Box>
-);
+const ProgressContent: FC<{ currentPhase: OfframpingPhase; currentPhaseIndex: number }> = ({
+  currentPhase,
+  currentPhaseIndex,
+}) => {
+  const [currentPercentage, setCurrentPercentage] = useState<number>(
+    Math.round((100 / numberOfPhases) * currentPhaseIndex),
+  );
+  const [targetPhaseIndex, setTargetPhaseIndex] = useState<number>(currentPhaseIndex);
+  const currentPhaseRef = useRef<OfframpingPhase>(currentPhase);
+  const currentPhaseIndexRef = useRef<number>(currentPhaseIndex);
+  const currentPhaseStartPercentage = useRef<number>((100 / numberOfPhases) * currentPhaseIndex);
+  const expectedEndTimeOfPhase = useRef<number>(Date.now() + OFFRAMPING_PHASE_SECONDS[currentPhase] * 1000);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
```

Comment:
Should we define at least a small timeout for the interval?

Reply:
This would be necessary if the flow completely stalls although there is no error (if there is an error, the page will automatically reload after 30 seconds or it will transition to the failure screen). In that case we could merely ensure that the user's browser will not require any more CPU, is that the purpose of this suggestion?

Reply:
Yes I was just concerned about the performance. It's probably okay to set the `timeout` to 100 or even 500 ms and still get a smooth transition. I'm referring to the `setInterval(f, timeout)` field if that's not clear.